### DISCUSSION TITLE
fieldArray input doesn't show default values

### DISCUSSION THREAD

**Discussion by MitchelKerckhaert** [react-hook-form/react-hook-form#7899] [Q&A] [answered]
Hi, I have been using react-hook-form for a couple months now, but this is the first time I have a use case for fieldArrays.

I have a form with one regular formfield and a fieldarray, both of which are initialized with a default value (a string and an array of strings respectively). However, the fieldArray input fields do not show the default values (but does register changes correctly). I have an example here https://codesandbox.io/s/wizardly-mopsa-je6ddd?file=/src/App.js. If anyone could help me understand what I am doing wrong that would be great!

**Comment by Moshyfawn** [comment, marked as answer]
Your field value shape is defined by your register function with the field's name
```js
{ ...register(`options.${fieldIdx}.value`)}
```
so
```js
{
  options: [
    {
      value: "Option One"
    },
    {
      value: "Option Two"
    }
  ]
}
```
which is the correct shape because RHF doesn't support flat arrays.

Now, your default value for options **is** a flat array and it doesn't correlate with the array field shape
```js
options: [
  'Option One',
  'Option Two'
]
```
Instead, it should be
```js
options: [
  {
    value: "Option One"
  },
  {
    value: "Option Two"
  }
]
```

**Reply by bluebill1049** [reply]
thanks @Moshyfawn that's right.

**Reply by MitchelKerckhaert** [reply]
ah thanks, I didn't realize flat arrays weren't supported. For anyone else who has a use case for flat arrays, you can map the result back to a flat array in the onsubmit function

**Reply by Moshyfawn** [reply]
Yep, exactly as intended, thx, @MitchelKerckhaert!

**Reply by AdnanDervisevic** [reply]
Why is that exactly as intended? What is the reasoning behind not supporting flat arrays? It seems like an obvious fit for this feature and has been discussed for years at this point (I found a thread from 2020 discussing this). It is seriously confusing to have something that handles arrays, but not flat arrays.

**Reply by Moshyfawn** [reply]
RHF needs to have a field UUID to correctly update and manipulate the array of fields. So, it has to be an object with the `id` field.
Please, refer to [RHF docs on useFieldArray](https://react-hook-form.com/api/usefieldarray/#rules) to learn more.
